src/dmd/json.d

```diff
@@ -968,3 +1062,20 @@ JsonFieldFlags tryParseJsonField(const(char)* fieldName)
     }
     return JsonFieldFlags.none;
 }
+
+/**
+Determines and returns the compiler interface which is one of `dmd`, `ldc`,
+`gdc` or `sdc`. Returns `null` if no interface can be determined.
+*/
+private const(char)* determineCompilerInterface()
+{
+    if (0 == strcmp(global.compiler.vendor, "Digital Mars D"))
```

Comment:
Pattern is `actual == expected`  (grep through the source code).
Of course `!strcmp` would work too.